urllib3/connectionpool.py

```diff
@@ -568,25 +568,22 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
             # Close the connection. If a connection is reused on which there
             # was a Certificate error, the next request will certainly raise
             # another Certificate error.
-            if conn:
-                conn.close()
-                conn = None
+            conn = conn and conn.close()
+            release_conn = True
```

Comment:
Hrm I could have sworn I left a reply but I guess it didn't take. —

How would the subsequent recursive call happen if we're raising an exception immediately after?